src/dodal/devices/undulator.py

```diff
@@ -63,3 +90,49 @@ def __init__(
                 self.length = None
 
         super().__init__(name)
+
+    @AsyncStatus.wrap
+    async def set(self, value: float):
+        await asyncio.gather(
+            self._set_undulator_gap(value),
+        )
+
+    async def _set_undulator_gap(self, energy_kev: float) -> None:
+        LOGGER.info(f"Setting DCM energy to {energy_kev:.2f} kev")
```

Comment:
Should: This is now decoupled from the DCM, we shouldn't be mentioning it in the logs/comments as it will cause confusion